nsq/reader.py

```diff
@@ -677,7 +669,7 @@ def _redistribute_rdy_state(self):
             # We also don't attempt to avoid the connections who previously might have had RDY 1
             # because it would be overly complicated and not actually worth it (ie. given enough
             # redistribution rounds it doesn't matter).
-            possible_conns = list(self.conns.values())
+            possible_conns = [c for c in conns if not (c.in_flight or c.rdy)]
```

Comment:
Sorry, I think I misunderstood our earlier discussion to have meant that that was the kind of case you weren't worried about.  The possibility you mention would be taken care of with:
```
            in_flight_or_rdy = sum(max(c.in_flight, c.rdy) for c in conns)
            if backoff_interval:
                max_in_flight = max(0, 1 - in_flight_or_rdy)
            else:
                max_in_flight = self.max_in_flight - in_flight_or_rdy
```
That's essentially the logic I proposed in https://github.com/nsqio/pynsq/issues/177#issuecomment-299917567.